layers/+intl/japanese/packages.el

```diff
@@ -0,0 +1,93 @@
+;;; packages.el --- Japanese Layer packages File for Spacemacs
+;;
+;; Copyright (c) 2012-2016 Sylvain Benner & Contributors
+;;
+;; Author: Sylvain Benner <sylvain.benner@gmail.com>
+;; URL: https://github.com/syl20bnr/spacemacs
+;;
+;; This file is not part of GNU Emacs.
+;;
+;;; License: GPLv3
+
+;; List of all packages to install and/or initialize. Built-in packages
+;; which require an initialization must be listed explicitly in the list.
+(setq japanese-packages
+      '(helm
+        migemo
+        avy-migemo
+        japanese-holidays
+        ddskk
+        pangu-spacing
+        org))
+
+(defun japanese/init-migemo ()
+  (use-package migemo
+    :config
+    (setq migemo-command "cmigemo")
+    (setq migemo-options '("-q" "--emacs" "-i" "\a"))
+    (setq migemo-dictionary "/usr/local/share/migemo/utf-8/migemo-dict")
+    (setq migemo-user-dictionary nil)
+    (setq migemo-regex-dictionary nil)
+    (setq migemo-coding-system 'utf-8-unix)
+    (migemo-init)
+    (with-eval-after-load "helm"
+      (helm-migemo-mode 1))
```

Comment:
AFAIK, `helm-migemo-mode` comes from `helm-migemo` package, not the `migemo`. So you have to add `helm-migemo` to `japanese-packages` and configure it via `japanese/init-helm-migemo` function.